src/main/webapp/utilities/MapsInterface.js

```diff
@@ -0,0 +1,95 @@
+class MapInterface {
+    /**
+     * @param element the DOM element the map to be rendered
+     * @param centerCoordinates coordinates object that will be the center of the map
+     * @param options (optional) additional parameters that might be needed
+     */
+    constructor(element, centerCoordinates, options = {}) {
+        this.map = new google.maps.Map(element, {
+            center: centerCoordinates,
+            zoom: options.zoom || 8,
+        });
+
+        // Service that fetches the route (used on drawRoute)
+        this.directionsService = options.directionsService || new google.maps.DirectionsService();
+
+        // Service that draws a route on the map (used on drawRoute)
+        this.directionsRenderer = options.directionsRenderer || new google.maps.DirectionsRenderer({
+                suppressMarkers: true, // do not display default markers of the route
+            });
+
+        this.directionsRenderer.setMap(this.map);
+
+        this.markers = [];
+        static const API_KEY = ''; // your API key
```

Comment:
Might be create JS file that contains only this constant, and exclude it from git repository?

Reply:
Can you have a look now?